### PR TITLE
server: report cpu cores quota to Prometheus

### PR DESCRIPTION
**Pull request by howardlau1999**
Signed-off-by: Howard Lau <howardlau1999@hotmail.com>

### What problem does this PR solve?

Problem Summary:

PD needs the CPU cores quota info of the TiKV server for auto-scaling.

### What is changed and how it works?

What's Changed:

The TiKV server will now report a new Gauge metric `tikv_server_cpu_cores_quota` to Prometheus. The metric value is set to `SysQuota::new().cpu_cores_quota()` on server startup.

### Check List 

Tests 

- No code

### Release note 

Reports CPU cores quota to Prometheus.

### CODE REVIEW

**Comment by howardlau1999** [comment]
We need this metric for making the auto-scaling decisions. Reporting it by Prometheus is simpler. Also, TiDB reports its CPU quota by Prometheus. We want to keep the query logic unified.

**Comment by rleungx** [comment]
@Connor1996 PTAL

File: src/server/metrics.rs [hunk 1]
```diff
@@ -332,6 +332,11 @@ lazy_static! {
             exponential_buckets(1f64, 5f64, 10).unwrap()
         )
         .unwrap();
+    pub static ref CPU_CORES_QUOTA: IntGauge = register_int_gauge!(
```
Comment (sticnarf) [inline src/server/metrics.rs:335]: I think a `Gauge` is better because CPU cores quota set in cgroup can be a fraction.

Comment (howardlau1999) [reply src/server/metrics.rs:335]: At the time of this PR, the SysQuota returns an integer quota. I will follow up on your changes.

**Comment by sticnarf** [comment]
LGTM

**Pull request merged**